ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectra.java

```diff
@@ -374,105 +376,142 @@ public void processDepositRequests(
     }
   }
 
-  /*
-   Implements process_consolidation from consensus-specs (EIP-7251)
-  */
+  /**
+   * Implements process_consolidation_request from consensus-spec (EIP-7251)
+   *
+   * @see <a
+   *     href="https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#new-process_consolidation_request"/>
+   */
   @Override
-  public void processConsolidations(
-      final MutableBeaconState state, final SszList<SignedConsolidation> consolidations)
-      throws BlockProcessingException {
+  public void processConsolidationRequests(
+      final MutableBeaconState state, final SszList<ConsolidationRequest> consolidationRequests) {
+    LOG.debug(
+        "process_consolidation_request: {} consolidation request to process from block at "
+            + "slot {}",
+        consolidationRequests.size(),
+        state.getSlot());
+
     final MutableBeaconStateElectra electraState = MutableBeaconStateElectra.required(state);
+    consolidationRequests.forEach(
+        consolidationRequest -> processConsolidationRequest(electraState, consolidationRequest));
+  }
+
+  private void processConsolidationRequest(
+      final MutableBeaconStateElectra state, final ConsolidationRequest consolidationRequest) {
+    final UInt64 slot = state.getSlot();
+    final UInt64 currentEpoch = miscHelpers.computeEpochAtSlot(slot);
```

Comment:
Maybe we could move this line further below right before where the `currentEpoch` gets used.